src/app/store/migrations.ts

```diff
@@ -3,16 +3,30 @@ import type { MigrationManifest, PersistedState } from 'redux-persist';
 import { toTokenSlug } from 'lib/assets';
 import { isCollectible } from 'lib/metadata';
 
+import { collectiblesMetadataInitialStateClone } from './collectibles-metadata/state';
 import type { RootState } from './root-state.type';
 
-type TypedPersistedRootState = Exclude<PersistedState, undefined> & RootState;
+import type { SLICES_BLACKLIST } from './index';
+
+type MakePropertiesOptional<T, K extends keyof T> = {
+  [P in keyof T]: P extends K ? T[P] | undefined : T[P];
+};
+
+/** Blacklisted slices are not rehydrated */
+type TypedPersistedRootState = Exclude<PersistedState, undefined> &
+  MakePropertiesOptional<RootState, typeof SLICES_BLACKLIST[number]>;
 
 export const MIGRATIONS: MigrationManifest = {
   '2': (persistedState: PersistedState) => {
     if (!persistedState) return persistedState;
     const typedPersistedState = persistedState as TypedPersistedRootState;
     const allTokensMetadata = typedPersistedState.tokensMetadata.metadataRecord;
 
+    // `collectiblesMetadata` slice data is absent. Setting initial value here.
+    // It is safe as it is blacklisted & won't be persisted in this (root) slice.
+    // @ts-expect-error // Due to the absence of `_persist` property yet
+    const collectiblesMetadata = (typedPersistedState.collectiblesMetadata = collectiblesMetadataInitialStateClone);
```

Comment:
It seems like these brackets are redundant here, since the assignment will still be from right to left